### PR TITLE
Creation of lineage for artifact and execution using D3.js

### PR DESCRIPTION
**Pull request by AyeshaSanadi**
# Related Issues / Pull Requests

List all related issues and/or pull requests if there are any.

# Description

Include a brief summary of the proposed changes.

# What changes are proposed in this pull request?

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected; for instance, 
      examples in this repository need to be updated too).
- [ ] This change requires a documentation update.

# Checklist:

- [ ] My code follows the style guidelines of this project (PEP-8 with Google-style docstrings).
- [ ] My code modifies existing public API, or introduces new public API, and I updated or wrote docstrings that
      uses Google-style formatting and any other formatting that is supported by mkdocs and plugins this project
      uses.
- [ ] I have commented my code.
- [ ] My code requires documentation updates, and I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

### CODE REVIEW

File: server/app/query_visualization_artifact_execution.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+            value = self.queue.popleft()
+            self.seen.remove(value)
+            return value
+        raise IndexError("dequeue from an empty queue")
+    
+    def __len__(self):
+        return len(self.queue)
+    
+    def __contains__(self, value):
+        return value in self.seen
```
Comment (shreyas-kulkarni09) [inline server/app/query_visualization_artifact_execution.py:28]: Is this class used anywhere? If yes, we can move this to some helper/util folder in separate file. We can add a doc string explaining functionality of the class .

**Review by shreyas-kulkarni09 — COMMENTED** [review]
A refactoring suggestion, if possible, let's move all "query" files to a particular folder.

**Comment by shreyas-kulkarni09** [comment]
> > A refactoring suggestion, if possible, let's move all "query" files to a particular folder.
> 
> Hi Shreyas, thank you for the feedback! I understand the suggestion to move all query files into one folder. However, moving them now would add code overhead and require extra testing to ensure everything works correctly. Since these are the final files with no further additions, keeping the current structure might be more efficient to avoid unnecessary changes. Let me know your thoughts!

Sure, for now we can have same file structure. We can have dedicated PR for this refactoring sometime later.

File: server/app/query_visualization_artifact_execution.py (hunk 1 above)
Comment (AyeshaSanadi) [reply server/app/query_visualization_artifact_execution.py]: Hi Shreyas, this is redundant code that we are not using, and we have now fixed it in the code. Thank you for catching that!

**Comment by varkha-d-sharma** [comment]
@annmary-roy this PR is ready to go from our side

**Review by annmary-roy — APPROVED** [review]

**Pull request merged**